src/tests/unit/cpp_wrapper/test_cpp_wrapper.cpp

```diff
@@ -112,6 +113,439 @@ test_cpp_wrapper_insert_on_all_implementations(
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, i, ret_val);
+	}
+}
+
+/**
+@brief	Tests insert/delete edge cases such as negative keys and big-digit keys.
+*/
+void
+test_cpp_wrapper_insert_update_edge_cases(
+		planck_unit_test_t *tc,
+		Dictionary<int, int> *dict
+) {
+	int ret_val;
+
+	dict->insert(-10, 3);
+	PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+	PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+
+	dict->update(-10, 30);
+	ret_val = dict->get(-10);
+	PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+	PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+	PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 30, ret_val);
+
+	dict->insert(1000, 99);
+	PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+	PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+
+	dict->update(1000, 9);
+	ret_val = dict->get(1000);
+	PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+	PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+	PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 9, ret_val);
+}
+
+/**
+@brief	Aggregate test to test insertion on all implementations.
+*/
+void
+test_cpp_wrapper_update_on_all_implementations(
+		planck_unit_test_t *tc
+) {
+	Dictionary<int, int> *dict;
+
+	dict = new BppTree<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int));
+	test_cpp_wrapper_insert_update(tc, dict);
+	test_cpp_wrapper_insert_update_edge_cases(tc, dict);
+	delete dict;
+
+	dict = new SkipList<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int), 7);
+	test_cpp_wrapper_insert_update(tc, dict);
+	test_cpp_wrapper_insert_update_edge_cases(tc, dict);
+	delete dict;
+
+	dict = new FlatFile<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int));
+	test_cpp_wrapper_insert_update(tc, dict);
+	test_cpp_wrapper_insert_update_edge_cases(tc, dict);
+	delete dict;
+
+	dict = new OpenAddressHash<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int), 50);
+	test_cpp_wrapper_insert_update(tc, dict);
+	test_cpp_wrapper_insert_update_edge_cases(tc, dict);
+	delete dict;
+
+	dict = new OpenAddressFileHash<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int), 50);
+	test_cpp_wrapper_insert_update(tc, dict);
+	test_cpp_wrapper_insert_update_edge_cases(tc, dict);
+	delete dict;
+}
+
+void
+test_cpp_wrapper_equality(
+		planck_unit_test_t	*tc,
+		Dictionary<int, int> *dict
+) {
+	int eq_key = 10;
+	int record_value = 0;
+
+	for(int i=0; i<eq_key; i++) {
+		dict->insert(eq_key, i);
+		PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+
+		dict->insert(i, i*2);
+		PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+	}
+
+	Cursor<int, int> *eq_cursor = dict->equality(eq_key);
+	PLANCK_UNIT_ASSERT_TRUE(tc, eq_cursor->hasNext());
+	eq_cursor->next();
+
+	while(eq_cursor->hasNext()) {
```

Comment:
This isn't actually the right way to write the Cursor loop - Check out https://github.com/iondbproject/iondb/blob/development/src/examples/iondb_example.ino#L39